src/common/tracer.h

```diff
@@ -0,0 +1,64 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+//forward declaration of req_state defined /rgw/rgw_common.h
+struct req_state;
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+class Jager_Tracer{
+  public:
+    Jager_Tracer(){}
+    ~Jager_Tracer(){
+      if(this->tracer)
+        this->tracer->Close();
+    }
+
+    void init_tracer(const char* tracerName,const char* filePath){
+        auto yaml = YAML::LoadFile(filePath);
+        auto configuration = jaegertracing::Config::parse(yaml);
+        this->tracer = jaegertracing::Tracer::make(
+        tracerName,
+        configuration,
+        jaegertracing::logging::consoleLogger());
+        opentracing::Tracer::InitGlobal(
+        std::static_pointer_cast<opentracing::Tracer>(tracer));
+    }
+    inline void finish_tracer(){
+      if(this->tracer)
+      	this->tracer->Close();
+    }
+    Span new_span(const char* spanName)const{
+      Span span=opentracing::Tracer::Global()->StartSpan(spanName);
+      return std::move(span);
+    }
+    Span child_span(const char* spanName,const Span& parentSpan)const{
+      Span span = opentracing::Tracer::Global()->StartSpan(spanName, {opentracing::ChildOf(&parentSpan->context())});
+      return std::move(span);
+    }
+    Span followup_span(const char *spanName, const Span& parentSpan)const{
+    Span span = opentracing::Tracer::Global()->StartSpan(spanName, {opentracing::FollowsFrom(&parentSpan->context())});
+    return std::move(span);
+  }
+private:
+  std::shared_ptr<opentracing::v2::Tracer> tracer = NULL;
+};
+
+// structure to manage spans to trace functions who have access to req_state in /rgw/*
+struct span_structure{
+    req_state* req_state_span = nullptr;
```

Comment:
you are not creating a span type here I guess, this might be misleading to making an assumption that this is a span type

Reply:
yeah you are right its misleading will change it

Reply:
`req_state_span` is actually a good name for that struct.
as for the data member, just call it `state` (in other places in the code we usually call it `s`)